src/render.rs

```diff
@@ -29,6 +32,7 @@ pub(crate) struct Admonition<'a> {
     pub(crate) directive: Directive,
     pub(crate) title: String,
     pub(crate) content: Cow<'a, str>,
+    pub(crate) css_id: Option<CssIdType>,
```

Comment:
We should lose the `Option` here, as each block will definitely have an id - we should know it at this point.
```suggestion
    pub(crate) css_id: CssIdType,
```

This will force some refactoring elsewhere, which should help move a few things around.